### PR TITLE
enh(install): add dependency mechanism to fresh installation

### PR DESCRIPTION
**Pull request by adr-mo**
## Description

Several issues occured during fresh installations of Centreon.
In fact sometimes the installation order is not correct (installing auto-discovery without license-manager and/or pp-manager).

This should not occur to prevent broken installations of modules.

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [ ] 19.04.x
- [x] 19.10.x
- [x] 20.04.x (master)

<h2> How this pull request can be tested ? </h2>

- Do a fresh installation of Centreon.
- On Step 8 if you only select auto-discovery to install, license-manager and pp-manager should also be installed.

**Important**

There is no message clearly indicating the dependencies. This will be handled later on the Extension -> Manager and in the long term when redoing the installation wizard. This is a **simple quickfix**

#### Community contributors & Centreon team

- [x] I followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

#### Centreon team only

- [ ] I have made sure that the **unit tests** related to the story are successful.
- [ ] I have made sure that **unit tests cover 80%** of the code written for the story.
- [ ] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: www/install/steps/process/process_step8.php [hunk 1]
```diff
@@ -44,14 +44,46 @@
     $utilsFactory = new \CentreonLegacy\Core\Utils\Factory($dependencyInjector);
     $utils = $utilsFactory->newUtils();
     $moduleFactory = new \CentreonLegacy\Core\Module\Factory($dependencyInjector, $utils);
+
     foreach ($parameters['modules'] as $module) {
+        /* If the selected module is already installed (as dependency for example)
+         * then we can skip the installation process
+         */
+        if ($result['modules'][$module]['install']) {
```
Comment (sc979) [inline www/install/steps/process/process_step8.php]:
```suggestion
        if (isset($result['modules'][$module]['install']) && $result['modules'][$module]['install'] === true) {
```

File: www/install/steps/process/process_step8.php [hunk 2]
```diff
@@ -44,14 +44,46 @@
...
+        $information = $moduleFactory->newInformation();
+        $moduleInformation = $information->getConfiguration($module);
+        /* if the selected module has dependencies defined in its configuration file
+         * then we need to install them before installing the selected module to
+         * ensure its correct installation
+         */
+        if (isset($moduleInformation['dependencies'])) {
+            foreach ($moduleInformation['dependencies'] as $dependency) {
+                // If the dependency is already installed skip install
+                if ($result['modules'][$dependency]['install']) {
```
Comment (sc979) [inline www/install/steps/process/process_step8.php]: isset and === true

Comment (adr-mo) [reply www/install/steps/process/process_step8.php]: done

File: www/install/steps/process/process_step8.php (hunk 1 above)
Comment (adr-mo) [reply www/install/steps/process/process_step8.php]: done thanks @sc979

File: www/install/steps/process/process_step8.php [hunk 3]
```diff
@@ -44,14 +44,48 @@
     $utilsFactory = new \CentreonLegacy\Core\Utils\Factory($dependencyInjector);
     $utils = $utilsFactory->newUtils();
     $moduleFactory = new \CentreonLegacy\Core\Module\Factory($dependencyInjector, $utils);
+
     foreach ($parameters['modules'] as $module) {
+        /* If the selected module is already installed (as dependency for example)
+         * then we can skip the installation process
+         */
+        if (isset($result['modules'][$module]['install']) &&
```
Comment (sc979) [inline www/install/steps/process/process_step8.php]:
```suggestion
        if (
            isset($result['modules'][$module]['install']) &&
```

File: www/install/steps/process/process_step8.php [hunk 4]
```diff
@@ -44,14 +44,48 @@
...
+            $result['modules'][$module]['install'] === true) {
```
Comment (sc979) [inline www/install/steps/process/process_step8.php]:
```suggestion
            && $result['modules'][$module]['install'] === true
        ) {
```

File: www/install/steps/process/process_step8.php [hunk 5]
```diff
@@ -44,14 +44,50 @@
...
+        $information = $moduleFactory->newInformation();
+        $moduleInformation = $information->getConfiguration($module);
+        /* if the selected module has dependencies defined in its configuration file
+         * then we need to install them before installing the selected module to
+         * ensure its correct installation
+         */
+        if (isset($moduleInformation['dependencies'])) {
+            foreach ($moduleInformation['dependencies'] as $dependency) {
+                // If the dependency is already installed skip install
+                if (isset($result['modules'][$dependency]['install']) &&
```
Comment (sc979) [inline www/install/steps/process/process_step8.php]:
```suggestion
                if (
                    isset($result['modules'][$dependency]['install'])
```

File: www/install/steps/process/process_step8.php [hunk 6]
```diff
@@ -44,14 +44,50 @@
...
+                    $result['modules'][$dependency]['install'] === true) {
```
Comment (sc979) [inline www/install/steps/process/process_step8.php]:
```suggestion
                    && $result['modules'][$dependency]['install'] === true
                ) {
```

**Pull request merged**